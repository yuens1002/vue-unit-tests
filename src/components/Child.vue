<template>
  <div>
    {msg}
  </div>
</template>

<script>
  export default {
    name: 'child-compoent',
    props: {
      msg: {
        required: true,
        type: String
      }
    }
  }
</script>

<style scoped>

</style>