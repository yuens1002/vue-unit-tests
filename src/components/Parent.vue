<template>
  <div id="parent">
    
    <div id="child" name="child-component"><child-component @custom="onCustom" :msg="msg" /></div>
    <p v-if="emitted">{{msg}}</p>
    <div><button @click="add()" name='my-button'>add one</button> {{count}}</div>
  </div>
</template>

<script>
  import Child from './Child'

  export default {
    name: 'ParentComponent',
    components: { ChildComponent: Child },
    data() {
      return {
        emitted: false,
        msg: 'hello',
        count: 0
      }
    },
    methods: {
      onCustom(value) {
        this.emitted = true
        this.msg = value
      },
      add () {
        this.count++
      }
    }
  }
</script>