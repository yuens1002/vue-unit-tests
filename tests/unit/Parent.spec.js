import { shallowMount } from '@vue/test-utils'
import Parent from '@/components/Parent'
import Child from '@/components/Child'

describe('Parent component', () => {
  const build = () => {
    const wrapper = shallowMount(Parent)
    return {
      wrapper,
      child: () => wrapper.find(Child)
    }
  }

  it('mounts itself', () => {
    expect(build().wrapper.html()).toMatchSnapshot()
    // expect(build().wrapper.attributes('id')).toBe('parent')
  })

  it('mounts the Child component', () => {
    const { child } = build()
    expect(child().exists()).toBe(true)
  })

  it('has an id "parent"', () => {
    const { wrapper } = build()
    expect(wrapper.attributes().id).toBe('parent')
  })

  it('has a div with an id "child" wrapped by another div', () => {
    const { wrapper } = build()
    expect(wrapper.contains('div > div#child')).toBe(true)
  })

  it('passes a props to the Child component', () => {
    const { child, wrapper } = build()
    wrapper.setData({ msg: 'this is a prop from parent' })
    expect(child().vm.msg).toBe('this is a prop from parent')
  })

  it('receives 2 emits from Child components', () => {
    const { child } = build()
    child().vm.$emit('custom', 'hello')
    child().vm.$emit('custom', 'hello2')
    expect(child().emittedByOrder().map(e => e.args)).toEqual([['hello'], ['hello2']])
  })

  it('add method should add 1', () => {
    const { wrapper } = build()
    const oldCount = wrapper.vm.count
    wrapper.find('button').trigger('click')
    expect(wrapper.vm.count).toBe(oldCount + 1)
  })
})
