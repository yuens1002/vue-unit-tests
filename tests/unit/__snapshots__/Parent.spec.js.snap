// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parent component mounts itself 1`] = `
<div id="parent">
  <div id="child" name="child-component">
    <child-component-stub msg="hello"></child-component-stub>
  </div>
  <!---->
  <div><button name="my-button">add one</button> 0</div>
</div>
`;
